### PR TITLE
Gh 518 function-equals-hashCode-toString

### PR DESCRIPTION
**Pull request by t616178**

### CODE REVIEW

File: core/function/pom.xml [hunk 1]
```diff
@@ -37,6 +37,12 @@
             <artifactId>jackson-databind</artifactId>
             <version>${jackson.version}</version>
         </dependency>
+
+        <dependency>
+            <groupId>commons-lang</groupId>
+            <artifactId>commons-lang</artifactId>
+            <version>2.6</version>
```
Comment (p013570) [inline core/function/pom.xml]: It would be better to use the commons-lang.version property.

**Comment by p013570** [comment]
The issue wants toString, equals and hashcode methods on all functions, the view and the schema. It looks like there are a few missing?

**Comment by t616178** [comment]
I've added the missing methods and made sure that properties from base classes are included in the methods (these were being excluded)

**Comment by p013570** [comment]
Changes look good. Just wondering how we force new Function implementations to override the toString, hashcode and equals methods? Could we add a test into ConsumerFunctionTest to somehow check these methods have been implemented and all public fields are included in these methods?

**Comment by t616178** [comment]
I think it would be better to just enforce that developers are expected to provide these methods if appropriate. I've updated the Wiki to reflect this.

**Comment by gaffer01** [comment]
Looks good. Should there be unit tests on the `equals()` methods? Or is that unnecessary?

**Pull request merged**